pkg/controller/build/build_controller.go

```diff
@@ -72,5 +78,116 @@ type ReconcileBuild struct {
 func (r *ReconcileBuild) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling Build")
+
+	b := &build.Build{}
+	err := r.client.Get(context.Background(), request.NamespacedName, b)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Validate if the spec.output.secretref exist in the namespace
+	if b.Spec.Output.SecretRef != nil {
+		if b.Spec.Output.SecretRef.Name != "" {
+			if err := r.validateOutputSecret(b.Spec.Output.SecretRef.Name, b.Namespace); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	}
+
+	// Validate if the build strategy is defined
+	if b.Spec.StrategyRef != nil {
+		if err := r.validateStrategyRef(b.Spec.StrategyRef, b.Namespace); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
 	return reconcile.Result{}, nil
 }
+
+func (r *ReconcileBuild) validateStrategyRef(s *build.StrategyRef, ns string) error {
+	switch *s.Kind {
+	case build.NamespacedBuildStrategyKind:
+		if err := r.validateBuildStrategy(s.Name, ns); err != nil {
+			return err
+		}
+	case build.ClusterBuildStrategyKind:
+		if err := r.validateClusterBuildStrategy(s.Name); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("unknown strategy %v", *s.Kind)
+	}
+	return nil
+}
+
+func (r *ReconcileBuild) validateBuildStrategy(n string, ns string) error {
+	list := &build.BuildStrategyList{}
+
+	if err := r.client.List(context.Background(), list, &client.ListOptions{Namespace: ns}); err != nil {
+		return errors.Wrapf(err, "listing BuildStrategies in ns %s failed", ns)
+	}
+
+	if len(list.Items) == 0 {
```

Comment:
Is it worth doing a `nil` check on `list` ?

Reply:
I think no, afaik if the call to get a [list](https://github.com/redhat-developer/build/pull/99/files#diff-1596a528bcf61af207bce1f85cb1202dR126) gives you nothing, the `len(list.Items)` should equal zero always. It should not panic, if this is the concern.

I did an example for validation, see https://play.golang.org/p/nP5rU1Yhyu4

Reply:
Thanks.